indico/modules/categories/controllers/management.py

```diff
@@ -102,15 +104,36 @@ def _process(self):
 
 class RHManageCategorySettings(RHManageCategoryBase):
     def _process(self):
-        defaults = FormDefaults(self.category,
-                                meeting_theme=self.category.default_event_themes['meeting'],
-                                lecture_theme=self.category.default_event_themes['lecture'])
+        additional_keys = ('google_wallet_application_credentials', 'google_wallet_issuer_name',
+                           'google_wallet_issuer_id')
+        if not config.ENABLE_GOOGLE_WALLET:
+            self.category.google_wallet_settings = {}
+        kwargs = {
+            'meeting_theme': self.category.default_event_themes['meeting'],
+            'lecture_theme': self.category.default_event_themes['lecture'],
+            'google_wallet_enabled': self.category.google_wallet_settings}
+        if self.category.google_wallet_settings:
+            kwargs.update({key: self.category.google_wallet_settings[key] for key in additional_keys})
+        defaults = FormDefaults(self.category, **kwargs)
         form = CategorySettingsForm(obj=defaults, category=self.category)
         icon_form = CategoryIconForm(obj=self.category)
         logo_form = CategoryLogoForm(obj=self.category)
 
         if form.validate_on_submit():
-            update_category(self.category, form.data, skip={'meeting_theme', 'lecture_theme'})
+            new_dict = form.data
+            new_dict.setdefault('google_wallet_settings', {})
+            if new_dict.pop('google_wallet_enabled'):
+                for key in additional_keys:
+                    new_dict['google_wallet_settings'][key] = new_dict.pop(key, {})
+                    if key == 'google_wallet_application_credentials':
+                        new_dict['google_wallet_settings'][key] = ast.literal_eval(
```

Comment:
`json.loads()` - this is json and not python, even if technically most valid JSON code will also be a valid Python dict literal.

you can also simply use a `JSONField` in the WTForm which will parse it for you...

Reply:
I tried json.loads but is returned me an error in a couple of cases, that's why I used literal_eval.

Reply:
Any examples of cases where this happened?

Reply:
I'll try again to see if I can raise it up. 
If I'll not be able to reproduce it, I'll use JSONField.

Reply:
Fixed using json.loads.